src/PotentialEnergyEquationTerms.jl

```diff
@@ -87,27 +132,56 @@ julia> model = NonhydrostaticModel(; grid, buoyancy, tracers);
 
 julia> geopotential_height = 0; # density variable will be σ₀
 
-julia> PotentialEnergy(model; geopotential_height)
+julia> PotentialEnergy(model)
 KernelFunctionOperation at (Center, Center, Center)
 ├── grid: 1×1×100 RectilinearGrid{Float64, Flat, Flat, Bounded} on CPU with 0×0×3 halo
 ├── kernel_function: g′z_ccc (generic function with 1 method)
-└── arguments: ("KernelFunctionOperation at (Center, Center, Center)", "KernelFunctionOperation at (Center, Center, Center)", "(g=9.80665, ρ₀=1020.0)")
+└── arguments: ("KernelFunctionOperation at (Center, Center, Center)", "(g=9.80665, ρ₀=1020.0)")
 ```
 """
-@inline function PotentialEnergy(model; geopotential_height = model_geopotential_height(model), location = (Center, Center, Center))
+@inline function PotentialEnergy(model; location = (Center, Center, Center),
+                                 geopotential_height = model_geopotential_height(model))
 
     validate_location(location, "PotentialEnergy")
-    validate_buoyancy(model.buoyancy)
+    validate_gravity_unit_vector(model.buoyancy.gravity_unit_vector)
+
+    return PotentialEnergy(model, model.buoyancy, geopotential_height)
+end
+
+@inline PotentialEnergy(model, buoyancy_model::NoBuoyancyModel, geopotential_height) =
+    throw(ArgumentError("Cannot calculate gravitational potential energy without a Buoyancy model."))
+
+@inline function PotentialEnergy(model, buoyancy_model::BuoyancyTracerModel, geopotential_height)
+
+    grid = model.grid
+    b = model.tracers.b
+
+    return KernelFunctionOperation{Center, Center, Center}(bz_ccc, grid, b)
+end
+
+@inline bz_ccc(i, j, k, grid, b) = b[i, j, k] * Zᶜᶜᶜ(i, j, k, grid)
+
+@inline function PotentialEnergy(model, buoyancy_model::BuoyancyLinearEOSModel, geopotential_height)
```

Comment:
```suggestion
@inline function potential_energy(model, buoyancy_model::BuoyancyLinearEOSModel, geopotential_height)
```

Reply:
What do you think @tomchor? I could make the change here then update the other functions in a new PR. Or could open a new PR to change functions to all be snake case in line with julia style. Now is as good a time as ever!?

Reply:
Let's keep the current style in this PR. We can discuss changes in an issue and have a separate PR to do it all at once if necessary.